src/test/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/DiscoverVariantsFromContigAlignmentsSAMSparkUnitTest.java

```diff
@@ -0,0 +1,144 @@
+                Arrays.asList( unAnnotatedVC ) }
+        );
+        tests.add(new Object[] {
+                Arrays.asList(
+                        new EvidenceTargetLink(
+                                new StrandedInterval(new SVInterval(0, 950, 1050), true),
+                                new StrandedInterval(new SVInterval(0, 1975, 2025), false),
+                                5, 7, new HashSet<>(), new HashSet<>())),
+                Arrays.asList( unAnnotatedVC ),
+                Arrays.asList( unAnnotatedVC, impreciseDeletion ) }
+        );
+        tests.add(new Object[] {
+                Arrays.asList(
+                        new EvidenceTargetLink(
+                                new StrandedInterval(new SVInterval(0, 190, 210), true),
+                                new StrandedInterval(new SVInterval(0, 310, 320), false),
+                                3, 4, new HashSet<>(), new HashSet<>()),
+                        new EvidenceTargetLink(
+                                new StrandedInterval(new SVInterval(0, 192, 215), true),
+                                new StrandedInterval(new SVInterval(0, 299, 303), false),
+                                2, 3, new HashSet<>(), new HashSet<>())),
+                Arrays.asList( unAnnotatedVC ),
+                Arrays.asList( annotatedVC ) }
+        );
+
+        return tests.toArray(new Object[][]{});
+    }
+
+    @Test(dataProvider = "evidenceTargetLinksAndVariants")
+    public void testProcessEvidenceTargetLinks(final List<EvidenceTargetLink> etls,
+                                               final List<VariantContext> inputVariants,
+                                               final List<VariantContext> expectedVariants) throws Exception {
+        final StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection params =
+                new StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection();
+
+        final PipelineOptions options = null;
+        final ReferenceMultiSource referenceMultiSource = new ReferenceMultiSource(options, twoBitRefURL, ReferenceWindowFunctions.IDENTITY_FUNCTION);
+
+        ReadMetadata metadata = Mockito.mock(ReadMetadata.class);
+        when(metadata.getMaxMedianFragmentSize()).thenReturn(300);
+
+
+        PairedStrandedIntervalTree<EvidenceTargetLink> evidenceTree = new PairedStrandedIntervalTree<>();
+        etls.forEach(e -> evidenceTree.put(e.getPairedStrandedIntervals(), e));
+
+        final List<VariantContext> processedVariantContexts =
+                DiscoverVariantsFromContigAlignmentsSAMSpark.processEvidenceTargetLinks(params, localLogger, evidenceTree,
+                        metadata, inputVariants, referenceMultiSource);
+
+        Assert.assertEquals(processedVariantContexts.size(), expectedVariants.size());
+        for (int i = 0; i < processedVariantContexts.size(); i++) {
```

Comment:
Seems that there's the `VariantContextTestUtils.assertVariantContextsAreEqual()` for use if you have no attribute to ignore.

Reply:
changed

Reply:
Sorry I gave the wrong name. But the `for` loop is no longer needed right?

Reply:
Whoops! Removed the loop.